stream.go

```diff
@@ -75,6 +78,7 @@ func newStream(session *Session, id uint32, state streamState) *Stream {
 		recvNotifyCh: make(chan struct{}, 1),
 		sendNotifyCh: make(chan struct{}, 1),
 		establishCh:  make(chan struct{}, 1),
+		closeChan:    make(chan struct{}, 1),
```

Comment:
closeChan is only ever closed, so there's no need for it to be buffered.

```suggestion
		closeChan:    make(chan struct{}),
```